lib/action_subscriber/middleware/error_handler.rb

```diff
@@ -8,25 +8,18 @@ def initialize(app)
       end
 
       def call(env)
-        job_mutex = ::Mutex.new
-        job_complete = ::ConditionVariable.new
+        @app.call(env)
+      rescue => error
+        logger.error "FAILED #{env.message_id}"
 
-        job_mutex.synchronize do
-          ::Thread.new do
```

Comment:
MarchHare already has a built-in handler to convert java io errors: https://github.com/ruby-amqp/march_hare/blob/d568180074c24616f1240904d0a98d435d847a81/lib/march_hare/exceptions.rb#L99-L117

They'll still be caught here, though:
```
jruby-9.1.12.0 :013 >   error = java.io.IOException.new("yolo")
 => java.io.IOException: yolo                                                                                                                                                                                                                                                             
jruby-9.1.12.0 :014 > raise error rescue puts "we're good"
we're good                                                                                                                                                                                                                                                                                
 => nil    
```